### PR TITLE
Ben/dots

### PR DESCRIPTION
**Pull request by benbeauchamp7**
Replaced the ASCII dots with components. Each dot has a link ID. Dots with the same linkID prop respond as a group. Click a dot will grey out all other blocks for which a dot with the same id does not exist- this is to make it easy to find corresponding dots with the same link ID.

When the scheduling algorithm is producing results, we can simply give each employee an ID and associative all dots for their section with that ID after scheduling.

### CODE REVIEW

File: src/Components/Misc/Dot.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+// Adds or removes a class from blocks containing dots with or without a certain link ID
+const modifyBlocks = (id: number, newClass: string, inverted: boolean = false, remove: boolean = false) => {
+  const selector = `div.dot${(inverted? ':not(' : '')}[link-id="${id}"]${(inverted? ')' : '')}`;
+  let linked = Array.from(document.querySelectorAll(selector));
+  if (remove) linked.forEach(e => e.parentElement?.parentElement?.classList.remove(newClass));
+  else linked.forEach(e => e.parentElement?.parentElement?.classList.add(newClass));
+}
+
+
```
Comment (dantebarbieri) [inline src/Components/Misc/Dot.tsx:25]: We could remove this whitespace.

File: src/assets/colors.json [hunk 2]
```diff
@@ -0,0 +1,105 @@
+{
+    "reference": "https://medialab.github.io/iwanthue/",
+    "colors": [
```
Comment (dantebarbieri) [inline src/assets/colors.json:3]: Seems like we'll need an arbitrarily large number of colors. There's a color space called L\*a\*b\* color which is meant to be that any change in any of the L, a, or b axes is always visually distinct. One such method, therefore, is to sample points at regular intervals in L\*a\*b\* and to convert the resulting colors to RGB to be stored in this file.

This removes the need to only have 100 colors references by another person and gives us the flexibility (if we so choose) to generate an arbitrary number of colors if we want to do it on-the-fly.

**Review by dantebarbieri — APPROVED** [review]
All in all, great work! I modified `dot.tsx` to `Dot.tsx` because both Windows and OSX collapse the case so the files 'should' be the same, but on my Windows 11 machine, Node failed to detect it when there was a case mismatch. That's my only meaningful change.

I offered a suggestion with [L\*a\*b\* color](https://en.wikipedia.org/wiki/CIELAB_color_space), but it is not necessary for now.

File: src/assets/colors.json (hunk 2 above)
Comment (benbeauchamp7) [reply src/assets/colors.json:3]: I think that's a great sprint 2 goal, I generated these colors essentially on a whim and I'm not terribly pleased with their color range so I was wanting something more robust later. We can probably generate these colors as a function of the link ID of the dot itself which would probably be pretty clean

**Pull request merged**